src/container/SequentialContainer/Deque.ts

```diff
@@ -342,36 +358,43 @@ class Deque<T> extends SequentialContainer<T> {
       arr.push(this.getElementByPos(i));
     }
     arr.sort(cmp);
-    for (let i = 0; i < this._length; ++i) this.setElementByPos(i, arr[i]);
+    for (let i = 0; i < this._length; ++i) {
+      this.setElementByPos(i, arr[i]);
+    }
+    return this;
   }
   /**
    * @description Remove as much useless space as possible.
    */
   shrinkToFit() {
     if (this._length === 0) return;
-    const arr: T[] = [];
-    this.forEach(function (el) {
-      arr.push(el);
-    });
-    this._bucketNum = Math.max(Math.ceil(this._length / this._bucketSize), 1);
-    this._length = this._first = this._last = this._curFirst = this._curLast = 0;
-    this._map = [];
-    for (let i = 0; i < this._bucketNum; ++i) {
-      this._map.push(new Array(this._bucketSize));
+    const newMap = [];
+    if (this._first === this._last) return;
+    else if (this._first < this._last) {
+      for (let i = this._first; i <= this._last; ++i) {
```

Comment:
It seems better to make `this._map` into a local variable, such as `const oldMap = this._map`,

access member variables(like `this._first`, `this._last`, `this._map` for each iteration is not good for performance.

Reply:
This part of the optimization can be done to the next minor version.